packages/edit-site/src/components/global-styles/border-panel.js

```diff
@@ -16,80 +16,90 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
-import { useSetting } from '../editor/utils';
+import { getSupportedGlobalStylesPanels, useSetting, useStyle } from './hooks';
 
 const MIN_BORDER_WIDTH = 0;
 
 // Defining empty array here instead of inline avoids unnecessary re-renders of
 // color control.
 const EMPTY_ARRAY = [];
 
-export function useHasBorderPanel( { supports, name } ) {
+export function useHasBorderPanel( name ) {
 	const controls = [
-		useHasBorderColorControl( { supports, name } ),
-		useHasBorderRadiusControl( { supports, name } ),
-		useHasBorderStyleControl( { supports, name } ),
-		useHasBorderWidthControl( { supports, name } ),
+		useHasBorderColorControl( name ),
+		useHasBorderRadiusControl( name ),
+		useHasBorderStyleControl( name ),
+		useHasBorderWidthControl( name ),
 	];
 
 	return controls.some( Boolean );
 }
 
-function useHasBorderColorControl( { supports, name } ) {
+function useHasBorderColorControl( name ) {
+	const supports = getSupportedGlobalStylesPanels( name );
 	return (
-		useSetting( 'border.customColor', name ) &&
+		useSetting( 'border.customColor', name )[ 0 ] &&
```

Comment:
Having to use [ 0 ] when we just need to retrieve a value is the main downside of the new API, but I guess we will need to live with that.

Reply:
To improve readability, maybe we could switch to a slightly more verbose syntax?

```js
// ...
const [ enabledInSettings ] = useSetting( 'border.customColor', name );
return enabledInSettings && supports.includes( 'borderColor' );
```

Reply:
Yes although more verbose, that syntax is more clear.